io/net/virt-net/lpm_basic.py

```diff
@@ -0,0 +1,266 @@
+
+
+class LPM(Test):
+    '''
+    Performs LPM from source server to remote server, and back.
+    '''
+    @skipUnless("ppc" in distro.detect().arch,
+                "supported only on Power platform")
+    @skipIf(IS_POWER_NV or IS_KVM_GUEST,
+            "This test is not supported on KVM guest or PowerNV platform")
+    def setUp(self):
+        '''
+        set up required packages and gather necessary test inputs
+        '''
+        self.install_packages()
+        self.rsct_service_start()
+
+        self.hmc_ip = self.get_mcp_component("HMCIPAddr")
+        if not self.hmc_ip:
+            self.cancel("HMC IP not got from lsrsrc command")
+        self.hmc_user = self.params.get("hmc_username", default='hscroot')
+        self.hmc_pwd = self.params.get("hmc_pwd", '*', default='abc123')
+
+	self.lpar = self.get_partition_name("Partition Name")
+        if not self.lpar:
+            self.cancel("LPAR Name not got from lsrsrc command")
+
+        self.login(self.hmc_ip, self.hmc_user, self.hmc_pwd)
+
+        cmd = 'lssyscfg -r sys -F name'
+        output = self.run_command(cmd)
+        self.server = ''
+
+	for sys in output:
+		cmd = 'lssyscfg -r lpar -F name -m %s' % sys 
+		lpars = self.run_command(cmd) 
+        	for lpar in lpars:
+            		if lpar in self.lpar:
+                		self.server = sys 
+                		break
+        if not self.server:
+            self.cancel("Managed System not got")
+
+        self.remote_server = self.params.get("remote_server", default=None)
+        if not self.remote_server:
+            self.cancel("No Remote Server specified for LPM")
+
+        if 'back' in str(self.name.name):
+            current_server = self.remote_server
+        else:
+            current_server = self.server
+
+        cmd = 'lssyscfg -r lpar -F name -m %s' % current_server
+        output = self.run_command(cmd)
+        for line in output:
+            if "%s-" % self.lpar in line:
+                self.lpar = line
+                break
+
+        if not self.is_lpar_in_server(current_server, self.lpar):
+            self.cancel("%s not in %s" % (self.lpar, current_server))
+
+    @staticmethod
+    def get_mcp_component(component):
+        '''
+        probes IBM.MCP class for mentioned component and returns it.
+        '''
+        for line in process.system_output('lsrsrc IBM.MCP %s' % component,
+                                          ignore_status=True, shell=True,
+                                          sudo=True).splitlines():
+            if component in line:
+                return line.split()[-1].strip('{}\"')
+        return ''
+
+    @staticmethod
+    def get_partition_name(component):
+        '''
+        get partition name from lparstat -i 
+        '''
+
+        output = process.system_output("lparstat -i", ignore_status=True,
+                                       shell=True, sudo=True)
+        if component in output:
```

Comment:
This can be done better, by checking the line in which `Partition Name` is present, and getting the partition name from that line